crates/huub/src/propagator/array_var_int_element.rs

```diff
@@ -19,20 +19,22 @@ use crate::{
 pub(crate) struct ArrayVarIntElementBounds {
 	vars: Vec<IntView>,                // Variables to be selected
 	y: IntView,                        // The selected variable
-	idx: IntView,                      // Variable that stores the index of the selected variable
+	index: IntView,                    // Variable that stores the index of the selected variable
 	action_list: Vec<(u32, IntEvent)>, // List of x variables that have been modified since the last propagation
+	min_support: TrailedInt,           // The minimum support of the selected variable
```

Comment:
Shouldn't the supports now be checked when you get notified? You only want to wake up if the current support has changed the relevant bound

Reply:
(or if the result has changed any bound of course)

Reply:
You are right. But some other propagation rules will be triggered in this case.  When a non-support variable `self.vars[i]` is updated, we still want to check whether `i` can be removed from the `self.index` variable. 

If we want to further optimize this by checking whether `i` is in the domain of `self.index` in event notification, we may need the `InspectionActions` instead of the `TrailingActions` to be passed to `notify_event`. Since this requires some underlying architecture changes, I don't include it in this pull request.

Reply:
Update the `notify_event` function to filter out unnecessary propagations